lockfile_generator/src/lib.rs

```diff
@@ -0,0 +1,124 @@
+use std::ffi::OsString;
+use std::path::{Path, PathBuf};
+use std::process::{Command, Stdio};
+use std::{fs, io};
+
+pub mod bundler;
+pub mod cargo;
+pub mod go;
+pub mod gradle;
+pub mod maven;
+pub mod npm;
+pub mod pip;
+pub mod poetry;
+pub mod python_requirements;
+pub mod yarn;
+
+/// Lockfile generation.
+pub trait Generator {
+    /// Lockfile file name.
+    fn lockfile_name(&self) -> &'static str;
+
+    /// Command for generating the lockfile.
+    fn command(&self) -> Command;
+
+    /// List of files conflicting with lockfile generation.
+    ///
+    /// These files are temporarily renamed during lockfile generation to ensure
+    /// the correct lockfile is updated.
+    fn conflicting_files(&self) -> Vec<&'static str> {
+        vec![self.lockfile_name()]
+    }
+
+    /// Generate the lockfile for a project.
+    ///
+    /// This will ignore all existing lockfiles and create a new lockfile based
+    /// on the current project configuration.
+    fn generate_lockfile(&self, project_path: &Path) -> Result<String> {
+        // Move files which interfere with lockfile generation.
+        let _relocators = self
+            .conflicting_files()
+            .drain(..)
+            .map(|file| FileRelocator::new(project_path.join(file)))
+            .collect::<Vec<_>>();
```

Comment:
This collects a `Vec<Result<Option<FileRelocator>>>` and doesn't do anything with potential errors (i.e., if the file could not be renamed).

Did you mean to collect a `Result<Vec<Option<FileRelocator>>>` so that you can propagate errors?

```suggestion
            .collect::<Result<Vec<_>>>()?;
```

Reply:
I don't think it's **super** important in this case, but yeah your suggestion was my intention and should be better.